src/evaluation/WorkspaceManager.jl

```diff
@@ -69,6 +69,9 @@ function start_relaying_logs((session, notebook)::Tuple{ServerSession, Notebook}
             next_log = take!(log_channel)
             putnotebookupdates!(session, notebook, UpdateMessage(:log, next_log, notebook))
         catch e
+            if !isopen(log_channel)
+                break
+            end
```

Comment:
`take!(log_channel)` is a blocking call, and we only call the `start_relaying_logs` functions once per workspace (async), so we should never `break` the loop, right?

Reply:
I noticed that when the user presses CTRL-C in the REPL to kill Pluto, the REPL is immediately flooded with an infinite stream of log messages:

```
┌ Error: Failed to relay log
│   exception =
│    Distributed.ProcessExitedException(2)
│    Stacktrace:
│     [1] worker_from_id(::Distributed.ProcessGroup, ::Int64) at /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.5/Distributed/src/cluster.jl:1074
│     [2] worker_from_id at /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.5/Distributed/src/cluster.jl:1071 [inlined]
│     [3] #remotecall_fetch#146 at /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.5/Distributed/src/remotecall.jl:421 [inlined]
│     [4] remotecall_fetch at /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.5/Distributed/src/remotecall.jl:421 [inlined]
│     [5] call_on_owner at /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.5/Distributed/src/remotecall.jl:494 [inlined]
│     [6] take! at /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.5/Distributed/src/remotecall.jl:626 [inlined]
│     [7] start_relaying_logs(::Tuple{Pluto.ServerSession,Pluto.Notebook}, ::Distributed.RemoteChannel{Channel{Any}}) at /home/chris/.julia/dev/Pluto/src/evaluation/WorkspaceManager.jl:69
│     [8] (::Pluto.WorkspaceManager.var"#1#2"{Pluto.ServerSession,Pluto.Notebook,Distributed.RemoteChannel{Channel{Any}}})() at ./task.jl:356
└                            @ Pluto.WorkspaceManager /home/chris/.julia/dev/Pluto/src/evaluation/WorkspaceManager.jl:72
```

This seemed to fix that case — I assumed because the remote process had been killed, and the `RemoteChannel` was now closed.  In this case, the async `Task` which is running `start_relaying_logs` is now an orphan, and we should allow it to exit (otherwise it'll run forever and never be cleaned up).

Reply:
As a side note - whenever you start code with `@async` it's worth thinking about how the `Task` will exit. Julia doesn't have a task cancellation API, so you need to do it manually by checking whether the resources needed by the task are still open. Typically these "resources" would be a channel of some kind.

See also https://github.com/JuliaLang/julia/issues/33248 for a lot of discussion about task lifetime.